gtk3/sample/gtk-demo/dialog.rb

```diff
@@ -1,137 +1,127 @@
+
+    vbox = Gtk::Box.new(:vertical, 8)
+    vbox.border_width = 8
+    frame.add(vbox)
+
+    # Standard message dialog
+    hbox = Gtk::Box.new(:horizontal, 8)
+    vbox.pack_start(hbox, :expand => false, :fill => false, :padding => 0)
+    button = Gtk::Button.new(:label => "_Message Dialog",
+                             :use_underline => true)
+    i = 0
+
+    button.signal_connect "clicked" do
+      dialog = Gtk::MessageDialog.new(:parent => window,
+                                      :flags => [:modal, :destroy_with_parent],
+                                      :type => :info,
+                                      :buttons => :ok_cancel,
+                                      :message => <<-MESSAGE)
+This message has been popped up the following
 number of times:
-
-#{@count}
-EOS
+MESSAGE
+      dialog.secondary_text = "#{i}"
       dialog.run
       dialog.destroy
-      @count += 1
+      i += 1
     end
 
-    def interactive_dialog_clicked
-      dialog = Gtk::Dialog.new('Interactive Dialog',
-                               self,
-                               Gtk::Dialog::MODAL |
-                               Gtk::Dialog::DESTROY_WITH_PARENT,
-                               [Gtk::Stock::OK, Gtk::ResponseType::OK],
-                               ["_Non-stock Button", Gtk::ResponseType::CANCEL]
-                               )
-
-      hbox = Gtk::Box.new(:horizontal, 0)
-      hbox.set_border_width(8)
-      dialog.vbox.pack_start(hbox, :expand => false, :fill => false, :padding => 0)
-
-      stock = Gtk::Image.new(Gtk::Stock::DIALOG_QUESTION, Gtk::IconSize::DIALOG)
-      hbox.pack_start(stock, :expand => false, :fill => false, :padding => 0)
-
-      table = Gtk::Table.new(2, 2, false)
-      table.set_row_spacings(4)
-      table.set_column_spacings(4)
-      hbox.pack_start(table, :expand => true, :fill => true, :padding => 0)
-      label = Gtk::Label.new('_Entry 1', true)
-      table.attach_defaults(label,
-                            0, 1, 0, 1)
+    hbox.pack_start(button, :expand => false, :fill => false, :padding => 0)
+    vbox.pack_start(Gtk::Separator.new(:horizontal),
+                    :expand => false,
+                    :fill => false,
+                    :padding => 0)
+
+    # Interactive dialog
+    hbox = Gtk::Box.new(:horizontal, 8)
+    vbox.pack_start(hbox, :expand => false, :fill => false, :padding => 0)
+    vbox2 = Gtk::Box.new(:vertical, 0)
+
+    button = Gtk::Button.new(:label => "_Interactive Dialog",
+                             :use_underline => true)
+
+    hbox.pack_start(vbox2, :expand => false, :fill => false, :padding => 0)
+    vbox2.pack_start(button, :expand => false, :fill => false, :padding => 0)
+
+    table = Gtk::Grid.new
+    table.row_spacing = 4
+    table.column_spacing = 4
+    hbox.pack_start(table, :expand => false, :fill => false, :padding => 0)
+
+    label = Gtk::Label.new("_Entry 1", :use_underline => true)
+    table.attach(label, 0, 0, 1, 1)
```

Comment:
Thanks. Could you try to do it? I want to see scripts that uses the API to confirm the API is really easy to use.